util/test/computePerfStats

```diff
@@ -1,232 +1,296 @@
+
+        # depending on whether we're asked to verify or reject
+        if not is_reject:
+            search_msg = "Checking for"
+            found_msg = "SUCCESS"
+            not_found_msg = "FAILURE"
+        else:
+            search_msg = "Checking for absence of"
+            found_msg = "FAILURE"
+            not_found_msg = "SUCCESS"
+        
+        # MATCH
+        if num: # if there's a line number
+            print("%s /%s/ on line %s... " % (search_msg, regex, num))
+            if re.match(regex_real, test_output[num_real]):
+                valid_output &= not is_reject
+                print(found_msg)
+            else:
+                valid_output &= is_reject
+                print(not_found_msg)
+        else: # no line number
+            print("%s /%s/ on any line... " % (search_msg, regex))
+            found = False
+            for line in test_output:
+                if re.match(regex_real, line):
+                    found = True
+                    print(found_msg)
+                    valid_output &= not is_reject
+                    break
+            if not found:
+                valid_output &= is_reject
+                print(not_found_msg)
+
+        if not valid_output:
+            print("Error: Invalid output found in %s" % args.test_output_file)
+    
+    return valid_output
+
+
+def read_key_file():
+    # read keys from .perfkeys (or other) file
+    verify_keys = []
+    keys = []
+    with open(args.keys_file, "r") as file:
+        for line in file:
+            key = line.strip()
+            if not key[0] == "#": # not a comment
+                if "verify" in key or "reject" in key:
+                    verify_keys.append(key)
+                else:
+                    keys.append(key)
+            else: # ignore comments unless they specify .dat
+                comment = key[1:].strip()
+                if comment[0:5] == "file:":
+                    global data_file
+                    data_file = "%s/%s" % (args.outputdir, comment.split()[1])
+
+    with open(errors_file, "a") as file:
+        file.write("processed %s\n" % args.keys_file)
+
+    return (keys, verify_keys)
+
+
+def create_data_file(keys):
+    # create and setup .dat file if not done already
+    if not os.path.isfile(data_file):
+        with open(errors_file, "a") as file:
+            file.write("created %s\n" % data_file)
+
+        with open(data_file, "a") as file:
+            file.write("# Date")
+            for key in keys:
+                file.write("\t" + key)
+            file.write("\n")
+    else:
+        verify_data_file(keys)
+
+
+def verify_data_file(keys_exp):
+    with open(data_file, "r") as file:
+        header = file.readline().strip().split("\t")
+        columns = len(header)
+        if "# Date" not in header[0]:
+            print("[Error: {} has no header.]".format(data_file))
+            exit(2)
+        keys_actual = header[1:] # get keys in header
+
+        for l in file:
```

Comment:
I think you need to skip comment and empty lines here

We automatically insert comments (I think in this file) for timeouts and sometimes manually insert comments ourselves